Services/Simulations.cs

```diff
@@ -169,31 +185,34 @@ public async Task<Models.Simulation> InsertAsync(Models.Simulation simulation, s
         /// </summary>
         public async Task<Models.Simulation> UpsertAsync(Models.Simulation simulation)
         {
-            if (simulation.Id != SIMULATION_ID)
+            if (string.IsNullOrEmpty(simulation.Id))
             {
-                this.log.Warn("Invalid simulation ID. Only one simulation is allowed");
-                throw new InvalidInputException("Invalid simulation ID. Use ID '" + SIMULATION_ID + "'.");
+                throw new InvalidInputException("Simulation ID is not specified.");
             }
 
-            var simulations = await this.GetListAsync();
-            if (simulations.Count > 0)
+            var existingSimulation = await this.GetAsync(simulation.Id);
+            if (existingSimulation != null)
             {
                 this.log.Info("Modifying simulation");
 
                 if (simulation.ETag == "*")
                 {
-                    simulation.ETag = simulations[0].ETag;
-                    this.log.Warn("The client used ETag='*' choosing to overwrite the current simulation");
+                    simulation.ETag = existingSimulation.ETag;
+                    this.log.Info("The client used ETag='*' choosing to overwrite the current simulation");
```

Comment:
pls make sure this is a Warning, because clients should never use "*" (that's a development feature only)